cmd/maestro/agent/cmd.go

```diff
@@ -47,53 +58,98 @@ func NewAgentCommand() *cobra.Command {
 	// workloadAgentOptions
-	fs.Int32Var(&maxJSONRawLength, "max-json-raw-length",
-		maxJSONRawLength, "The maximum size of the JSON raw string returned from status feedback")
-	fs.DurationVar(&agentOption.StatusSyncInterval, "status-sync-interval",
-		agentOption.StatusSyncInterval, "Interval to sync resource status to hub")
-	fs.DurationVar(&agentOption.AppliedManifestWorkEvictionGracePeriod, "resource-eviction-grace-period",
-		agentOption.AppliedManifestWorkEvictionGracePeriod, "Grace period for resource eviction")
-	fs.StringVar(&commonOptions.SpokeClusterName, "consumer-name",
-		commonOptions.SpokeClusterName, "Name of the consumer")
+	fs.Float32Var(&o.CommonOptions.CommoOpts.QPS, "kube-api-qps",
+		o.CommonOptions.CommoOpts.QPS, "QPS to use while talking with apiserver on spoke cluster")
+	fs.IntVar(&o.CommonOptions.CommoOpts.Burst, "kube-api-burst",
+		o.CommonOptions.CommoOpts.Burst, "Burst to use while talking with apiserver on spoke cluster")
+	fs.Int32Var(&o.WorkOptions.MaxJSONRawLength, "max-json-raw-length",
+		o.WorkOptions.MaxJSONRawLength, "The maximum size of the JSON raw string returned from status feedback")
+	fs.DurationVar(&o.WorkOptions.StatusSyncInterval, "status-sync-interval",
+		o.WorkOptions.StatusSyncInterval, "Interval to sync resource status to hub")
+	fs.DurationVar(&o.WorkOptions.AppliedManifestWorkEvictionGracePeriod, "resource-eviction-grace-period",
+		o.WorkOptions.AppliedManifestWorkEvictionGracePeriod, "Grace period for resource eviction")
+	fs.StringVar(&o.CommonOptions.SpokeClusterName, "consumer-name",
+		o.CommonOptions.SpokeClusterName, "Name of the consumer")
 	// message broker config file
-	fs.StringVar(&agentOption.WorkloadSourceConfig, "message-broker-config-file",
-		agentOption.WorkloadSourceConfig, "The config file path of the message broker, it can be mqtt broker or kafka broker")
-	fs.StringVar(&agentOption.WorkloadSourceDriver, "message-broker-type", "mqtt", "Message broker type (default: mqtt)")
-	fs.StringVar(&agentOption.CloudEventsClientID, "agent-client-id",
-		agentOption.CloudEventsClientID, "The ID of the agent client, by default it is <consumer-id>-work-agent")
-	fs.StringSliceVar(&agentOption.CloudEventsClientCodecs, "agent-client-codecs",
-		[]string{"manifest"}, "The codecs of the agent client. The valid codecs are manifest and manifestbundle")
+	fs.StringVar(&o.WorkOptions.WorkloadSourceConfig, "message-broker-config-file",
+		o.WorkOptions.WorkloadSourceConfig, "The config file path of the message broker, it can be mqtt broker or kafka broker")
+	fs.StringVar(&o.WorkOptions.WorkloadSourceDriver, "message-broker-type",
+		o.WorkOptions.WorkloadSourceDriver, "Message broker type")
+	fs.StringVar(&o.WorkOptions.CloudEventsClientID, "agent-client-id",
+		o.WorkOptions.CloudEventsClientID, "The ID of the agent client, by default it is <consumer-id>-work-agent")
+	fs.StringSliceVar(&o.WorkOptions.CloudEventsClientCodecs, "agent-client-codecs",
```

Comment:
can we propagate that up to the agent template https://github.com/openshift-online/maestro/blob/e3bdb59e5444b1e87b0a28828e0ae59d034f010c/templates/agent-template.yml ?